janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/addedrelations/ConcurrentAddedRelations.java

```diff
@@ -37,11 +41,17 @@ public synchronized boolean remove(final InternalRelation relation) {
 
     @Override
     public synchronized Iterable<InternalRelation> getView(final Predicate<InternalRelation> filter) {
-        return super.getView(filter);
+        return copyView(super.getView(filter));
     }
 
     @Override
     public synchronized Collection<InternalRelation> getAll() {
-        return super.getAll();
+        return new ArrayList<>(super.getAll());
+    }
```

Comment:
I see `getAll()` method is used in a single place (to pass added relations into the commit logic).
I'm thinking concurrent access to this collection could happen only in the situation when we are in the process of committing the transaction but also another thread is adding new relations into this transaction. 
I'm can't think about cases when adding new relations while committing the transaction makes sense. I could miss some picture here and could be that this part might be needed for future optimizations / features, but I just can't think about the case when wrapping this collection into mutable collection is needed. Right now `super.getAll()` returns unmodifiable collection which can't be mutated by the caller.
Perhaps you could give some example why this change is needed.

Reply:
The only usage I found when `getAll()` is called is when we commit the transaction. I that case, I believe, only a single thread will be committing a transaction. I'm not sure committing the same transaction OR mutating elements when another thread is committing a transaction is a safe operation.
I do understand that the class is called `ConcurrentAddedRelations` and it should be assumed that all method under this class will return thread-safe collections, but it seems it's an unnecessary operation in this case.
What I was thinking is that we can change the naming of this interface method from `getAll()` to `getAllUnsafe()` which would clarify to the caller that the product of this method is not thread-safe.

However, I'm OK if you want to leave it as is because this method is called only during the transaction commit operation. Thus, shouldn't be expensive.

Reply:
I have renamed the method to `getAllUnsafe` and removed the copy